apps/site/assets/ts/models/alert.ts

```diff
@@ -19,6 +20,92 @@ export const alertsByStop = (alerts: Alert[], stopId: StopId): Alert[] =>
       !!entities.stop && entities.stop!.some((id: StopId) => id === stopId)
   );
 
+const hasEffect = (alerts: Alert[], effect: string): boolean =>
+  some(alerts, alert => alert.effect === effect);
+
+export const isSuspension = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "suspension");
+
+export const isShuttleService = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "shuttle");
+
+export const isDetour = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "detour");
+
+export const alertsWithStop = (alerts: Alert[]): Alert[] =>
+  alerts.filter(
+    ({ informed_entity: entites }: Alert): boolean => !!entites.stop
+  );
+
+export const routeWideAlerts = (alerts: Alert[]): Alert[] =>
+  alerts.filter(({ informed_entity: { entities } }: Alert): boolean =>
+    entities.some(entity => !entity.stop && !entity.trip)
+  );
+
+export const alertsByRoute = (alerts: Alert[]): { [key: string]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      // create a map of every route id to the alert
+      const explodedAlerts = reduce(
+        alert.informed_entity.route,
+        (innerAcc, routeId) => {
+          return { ...innerAcc, [routeId]: [alert] };
+        },
+        {}
+      );
+
+      return mergeWith(acc, explodedAlerts, (objValue, srcValue) => {
+        if (isArray(objValue)) {
+          return objValue.concat(srcValue);
+        }
+        return srcValue;
+      });
+    },
+    {}
+  );
+};
+
+// This will only return alerts with direction id specified (aka affecting a single direction)
+export const alertsByDirectionId = (
+  alerts: Alert[]
+): { [key: number]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      const explodedAlerts = reduce(
+        alert.informed_entity.direction_id,
+        (innerAcc, directionId) => {
+          // only include specified directions
+          if (directionId !== null) {
+            return { ...innerAcc, [directionId]: [alert] };
```

Comment:
question: similar to the comment on `alertsByRoute`, is there a test case for when there are multiple alerts in a direction?
This line reads to me like there will only ever be one alert per directionId, but I may be overlooking something.

Reply:
Yeah this function (and `alertsByRoute` is a bit complicated.  What this function does is create an object with directionId as the key, and for a value an array of alerts affecting that route.
Yeah are right in thinking that this part of the code returns a single alert for directionId.

This inner `reduce` takes a single alert, and creates an object with each directionId it maps the alert to it.
Ex
`const alert = { id: 1, informed_entity: {direction_id: [0, 1]}};`
Becomes:
```
{
  0: [{id: 1, informed_entity: {direction_id: [0, 1]}],
  1: [{id: 1, informed_entity: {direction_id: [0, 1]}}]
}
  ```
 
The 2nd part of the function below is where the combination happens.  The `mergeWith` function takes the result of the inner `reduce` (explodedAlerts), and combines with the accumulator from the outer `reduce` (starts as an empty object {}).  This will combine the 2 objects and merge the array values of the matching keys.  So example:
```
{
  0: [{id: 1, informed_entity: {direction_id: [0, 1]}],
  1: [{id: 1, informed_entity: {direction_id: [0, 1]}}]
}

{
  0: [{id: 2, informed_entity: {direction_id: [0, 1]}],
  1: [{id: 2, informed_entity: {direction_id: [0, 1]}}]
}
  ```
  Becomes:
  ```
{
  0: [{id: 1, informed_entity: {direction_id: [0, 1]}, {id: 2, informed_entity: {direction_id: [0, 1]}],
  1: [{id: 1, informed_entity: {direction_id: [0, 1]}}, {id: 2, informed_entity: {direction_id: [0, 1]}]
}
```
after the merge with.
This last object is what is returned from the function

Reply:
Also there isn't a test that has more than one alert for the direction.  I will modify the test to include one.  Though this function follows the same logic as the `alertsByRoute` which does have a test for multiple alerts for the route.

Reply:
I see, thanks for explaining! I was getting tripped up in the nested reduces and keeping track of what was being accumulated at each level. May be worth considering some of the same recommendations applied from the comment on the alertsByRoute thread here. 

For example, perhaps this function could be simplified to  (pseudocode)
```
chain(alerts)
.flatmap(alert => alert.informed_entity.direction_id.map(direction_id => {direction_id, alert))
.groupBy((alert) => alert.direction_id)
.value()
```